### PR TITLE
[O2B-829] run number instead of run id in run details

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- existing endpoints which used `:runId`, use `:runNumber` now instead
- endpoint `GET /api/runs/:runId` moved to `GET /api/legacy/runs/:runId`
- `run-detail` page use `runNumber` instead of `runId` as run identifier, using `runId` is still possible but not recommended 
 
Notable changes for developers:
- tests update
- [runs.controller.js](https://github.com/AliceO2Group/Bookkeeping/pull/1219/files#diff-b83a620986f3ddfce92ac362116d4d019b9f8460f207beecd92d87bcbeebf6ba) refactor

Changes made to the database:
- NA

### CODE REVIEW

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/Runs/Details/RunDetailsModel.js [hunk 1]
```diff
@@ -35,13 +35,16 @@ export class RunDetailsModel extends Observable {
     /**
      * Constructor
      *
-     * @param {number} runId the id of the run to display
+     * @param {number} [runIdentifier.runNumber] the id of the run to display
+     * @param {number} [runIdentifier.runId] this paramter is deprecated, use runNumber instead
      * @param {string|null|undefined} [panelKey=null] the key of the current panel
      */
-    constructor(runId, panelKey) {
+    constructor({ runNumber, runId }, panelKey) {
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/Runs/Details/RunDetailsModel.js:42]: applied

File: lib/public/Model.js [hunk 2]
```diff
@@ -185,8 +185,7 @@ export default class Model extends Observable {
                 }
                 break;
             case 'run-detail':
-                this.runs.loadDetails(parseInt(this.router.params.id, 10), this.router.params.panel);
-
+                this.runs.loadDetails(this.router.params, this.router.params.panel);
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/Model.js:188]: applied

File: lib/usecases/run/GetLogsByRunUseCase.js [hunk 3]
```diff
@@ -35,10 +35,11 @@ class GetLogsByRunUseCase {
      */
     async execute(dto) {
         const queryBuilder = new QueryBuilder();
-        const { params } = dto;
-        const { runId } = params;
+        const { params: { runNumber } } = dto;
 
-        queryBuilder.where('id').is(runId);
+        if (runNumber) {
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/usecases/run/GetLogsByRunUseCase.js]: right, applied

File: lib/server/controllers/runs.controller.js [hunk 4]
```diff
@@ -126,7 +131,7 @@ const getRunById = async (request, response, next) => {
  *                        next middleware function.
  * @returns {undefined}
  */
-const getLogsByRunId = async (request, response, next) => {
+const getLogsByRunIdentifier = async (request, response, next) => {
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/server/controllers/runs.controller.js]: right, renamed

File: lib/server/controllers/runs.controller.js [hunk 5]
```diff
@@ -36,6 +36,18 @@ const {
 const { dtoValidator } = require('../utilities');
 const { ApiConfig } = require('../../config/index.js');
 
+// eslint-disable-next-line require-jsdoc
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/server/controllers/runs.controller.js]: added

File: lib/public/views/Runs/Details/RunDetailsModel.js [hunk 6]
```diff
@@ -73,15 +76,36 @@ export class RunDetailsModel extends Observable {
...
+    get runNumber() {
+        return this._runNumber;
+    }
+
     /**
      * Define the id of the run to fetch and fetch the related data
      *
+     * @param {number} runNumber the id of the run to display
+     */
+    set runNumber(runNumber) {
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/Runs/Details/RunDetailsModel.js]: refactored

File: lib/public/views/Runs/Runs.js [hunk 7]
```diff
@@ -81,17 +81,26 @@ export default class RunModel extends Observable {
 
     /**
      * Load the details page for the given run
-     *
-     * @param {number} runId  the id of the run to display
+     * @param {number|string|undefined} [runIdentifier.runNumber]  the runNumber of the run to display
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/Runs/Runs.js]: applied

File: lib/public/views/Runs/Runs.js [hunk 8]
```diff
@@ -81,17 +81,26 @@ export default class RunModel extends Observable {
...
+            this._detailsModel = new RunDetailsModel({ runId, runNumber }, panelKey);
             this._detailsModel.bubbleTo(this);
         } else {
-            this._detailsModel.runId = runId;
+            if (runNumber) {
+                this._detailsModel.runNumber = runNumber;
+            } else if (runId) {
+                this._detailsModel.runId = runId;
+            } else {
+                throw new Error('No run identifier to fetch its details');
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/Runs/Runs.js]: added

File: lib/public/views/Runs/Details/RunDetailsModel.js [hunk 9]
```diff
@@ -333,14 +315,11 @@ export class RunDetailsModel extends Observable {
...
-     * @param {number} runNumber the run number of the current run
-     * @param {string|null|undefined} [panelKey=null] the current panel key (null will use default)
      */
-    constructor(runNumber, panelKey) {
+    constructor() {
         super(Object.values(RUN_DETAILS_PANELS_KEYS));
-        this._runNumber = runNumber;
-        this.currentPanelKey = panelKey;
+        this._runNumber = null;
+        this.currentPanelKey = 'logs';
```
Comment (martinboulais) [inline lib/public/views/Runs/Details/RunDetailsModel.js:322]: Never use magic strings, you always have constants. Probably that the currentPanelKey is already defining default panelKey at this point, else you should either not define it at all or set it to null? What was the original behavior?

File: test/lib/usecases/run/UpdateRunUseCase.test.js [hunk 10]
```diff
@@ -159,14 +159,14 @@ module.exports = () => {
         });
 
         it('should successfully retrieve run via ID, store and return the new run with eorReasons passed as to update fields', async () => {
```
Comment (martinboulais) [inline test/lib/usecases/run/UpdateRunUseCase.test.js:161]: Please also rename the tests titles!

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/Runs/Details/RunDetailsModel.js (hunk 9 above)
Comment (xsalonx) [reply lib/public/views/Runs/Details/RunDetailsModel.js]: refactored

File: test/lib/usecases/run/UpdateRunUseCase.test.js (hunk 10 above)
Comment (xsalonx) [reply test/lib/usecases/run/UpdateRunUseCase.test.js]: renamed

File: lib/public/views/Runs/Details/RunDetailsModel.js (hunk 9 above)
Comment (martinboulais) [reply lib/public/views/Runs/Details/RunDetailsModel.js]: Please read again my comment, and check how the code works!

**Review by martinboulais — APPROVED** [review]

**Pull request merged**